File: pokedex.py
## -*- coding: utf-8 -*-
import json

class Pokedex:
    id = 0
    name = ""
    classification = ""
    height = 0
    weight = 0
    description = ""

    def __init__(self):
        self.pokedex_url = "./pokedex/pokedex.json"
    
    def exchange_gen(self, no, gen, minor):
        exchange_url = "./pokedex/gen" + str(gen) + "/gen" + str(gen) + "_no.json"
        try:
            with open(exchange_url, encoding='utf-8') as f:
                exchange_json = json.load(f)
            if minor:
                # マイナーチェンジ
                if str(no) in exchange_json[1]:
                    result = exchange_json[1][str(no)]
                else:
                    result = 0
            else:
                if str(no) in exchange_json[1]:
                    result = exchange_json[0][str(no)]
                else:
                    result = 0
        except:
            result = 0
        return int(result)

    # 読み込むポケモンの全国図鑑Noと世代を指定
    # def load(self, no, gen):
    def load(self, no, gen, minor):
        # 世代指定が0の場合は全国図鑑を探す
        if gen > 0:
            no = self.exchange_gen(no, gen, minor)
        # pokedex.jsonからの情報をロード
        if no > 0:
            with open(self.pokedex_url,encoding='utf-8') as f:
                pokedex_json = json.load(f)
            self.id = pokedex_json[no - 1]['id']
            self.name = pokedex_json[no - 1]['name']
            self.classification = pokedex_json[no - 1]['classification']
            self.height = pokedex_json[no - 1]['height']
            self.weight = pokedex_json[no - 1]['weight']

            gen_url = "./pokedex/gen" + str(gen) + "/gen" + str(gen) + ".json"
            with open(gen_url,encoding='utf-8') as f:
                gen_json = json.load(f)

pokedex = Pokedex()
pokedex.load(1000, 7, False)

print(pokedex.name)

File: pokedex.rb
require 'sqlite3'
require 'json'

# SQLiteの文字列をエスケープする関数
def escape_sql_string(str)
    return '' if str.nil?
    str.to_s.gsub("'", "''")
end

sqlite = SQLite3::Database.open 'pokedex.db'

# Start transaction for table creation
sqlite.transaction
sql = <<SQL
DROP TABLE IF EXISTS pokedex;
SQL
sqlite.execute(sql)

sql = <<SQL
CREATE TABLE pokedex (
    id INTEGER,
    form TEXT,
    region TEXT,
    jpn TEXT,
    eng TEXT,
    ger TEXT,
    fra TEXT,
    kor TEXT,
    chs TEXT,
    cht TEXT,
    classification TEXT,
    height TEXT,
    weight TEXT
);
SQL
sqlite.execute(sql)
# sqlite.commit

pokedex = JSON.parse(File.read('./pokedex/pokedex.json'))
pokedex["pokedex"].each do |pokemon|
    puts pokemon
    pokemon[""].each do |form|
        puts form
        sql = <<SQL
INSERT INTO pokedex
VALUES (
    #{pokemon['id']},
    '#{escape_sql_string(form['form'])}',
    '#{escape_sql_string(form['region'])}',
    '#{escape_sql_string(pokemon['name']['jpn'])}',
    '#{escape_sql_string(pokemon['name']['eng'])}',
    '#{escape_sql_string(pokemon['name']['ger'])}',
    '#{escape_sql_string(pokemon['name']['fra'])}',
    '#{escape_sql_string(pokemon['name']['kor'])}',
    '#{escape_sql_string(pokemon['name']['chs'])}',
    '#{escape_sql_string(pokemon['name']['cht'])}',
    '#{escape_sql_string(form['classification'])}',
    '#{escape_sql_string(form['height'])}',
    '#{escape_sql_string(form['weight'])}'
);
SQL
        sqlite.execute(sql)
    end

    if pokemon["region_form"]
        pokemon["region_form"].each do |form|
            puts form
            sql = <<SQL
INSERT INTO pokedex
VALUES (
    #{pokemon['id']},
    '#{escape_sql_string(form['form'])}',
    '#{escape_sql_string(form['region'])}',
    '#{escape_sql_string(pokemon['name']['jpn'])}',
    '#{escape_sql_string(pokemon['name']['eng'])}',
    '#{escape_sql_string(pokemon['name']['ger'])}',
    '#{escape_sql_string(pokemon['name']['fra'])}',
    '#{escape_sql_string(pokemon['name']['kor'])}',
    '#{escape_sql_string(pokemon['name']['chs'])}',
    '#{escape_sql_string(pokemon['name']['cht'])}',
    '#{escape_sql_string(form['classification'])}',
    '#{escape_sql_string(form['height'])}',
    '#{escape_sql_string(form['weight'])}'
);
SQL
            sqlite.execute(sql)
        end
    end

    if pokemon["mega_evolution"]
        pokemon["mega_evolution"].each do |form|
            sql = <<SQL
INSERT INTO pokedex
VALUES (
    #{pokemon['id']},
    '#{escape_sql_string(form['form'])}',
    '#{escape_sql_string(form['region'])}',
    '#{escape_sql_string(form['name']['jpn'] || pokemon['name']['jpn'])}',
    '#{escape_sql_string(form['name']['eng'] || pokemon['name']['eng'])}',
    '#{escape_sql_string(form['name']['ger'] || "")}',
    '#{escape_sql_string(form['name']['fra'] || "")}',
    '#{escape_sql_string(form['name']['kor'] || "")}',
    '#{escape_sql_string(form['name']['chs'] || "")}',
    '#{escape_sql_string(form['name']['cht'] || "")}',
    '#{escape_sql_string(form['classification'])}',
    '#{escape_sql_string(form['height'])}',
    '#{escape_sql_string(form['weight'])}'
);
SQL
            puts sql
            sqlite.execute(sql)
        end
    end

    if pokemon["gigantamax"]
        pokemon["gigantamax"].each do |form|
            sql = <<SQL
INSERT INTO pokedex
VALUES (
    #{pokemon['id']},
    '#{escape_sql_string(form['form'])}',
    '#{escape_sql_string(form['region'])}',
    '#{escape_sql_string(form['name']['jpn'] || pokemon['name']['jpn'])}',
    '#{escape_sql_string(form['name']['eng'] || pokemon['name']['eng'])}',
    '#{escape_sql_string(form['name']['ger'] || "")}',
    '#{escape_sql_string(form['name']['fra'] || "")}',
    '#{escape_sql_string(form['name']['kor'] || "")}',
    '#{escape_sql_string(form['name']['chs'] || "")}',
    '#{escape_sql_string(form['name']['cht'] || "")}',
    '#{escape_sql_string(form['classification'])}',
    '#{escape_sql_string(form['height'])}',
    '#{escape_sql_string(form['weight'])}'
);
SQL
            sqlite.execute(sql)
        end
    end
end
sqlite.commit